### PR TITLE
CORE-344: Add endpoint for hd case diary

### PR DESCRIPTION
**Pull request by marcoebbinghaus**
Added endpoint to get a diary / list of diary entries for a given hd case.

### CODE REVIEW

**Comment by marcoebbinghaus** [comment]
Thank you, I added the diary link but didn't add the trailing slashes according to our new formatting rules which changed contemporaneous to this PR.

File: backend/src/main/java/quarano/department/web/TrackedCaseDiaryEntrySummary.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * @author Marco Ebbinghaus
+ */
+public class TrackedCaseDiaryEntrySummary extends RepresentationModel<TrackedCaseDiaryEntrySummary> {
+
+	private final DiaryEntry entry;
+	private final MessageSourceAccessor messages;
```
Comment (jekutzsche) [inline backend/src/main/java/quarano/department/web/TrackedCaseDiaryEntrySummary.java:23]: `Messages` isn't used! Shouldn't the dependence then be better removed?

File: backend/src/main/java/quarano/department/web/TrackedCaseDiaryEntrySummary.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	private final MapperWrapper mapper;
+
+	public TrackedCaseDiaryEntrySummary(DiaryEntry diaryEntry, MessageSourceAccessor messages, MapperWrapper mapper) {
+		this.entry = diaryEntry;
+		this.messages = messages;
+		this.mapper = mapper;
+	}
```
Comment (jekutzsche) [inline backend/src/main/java/quarano/department/web/TrackedCaseDiaryEntrySummary.java:30]: Why do you use an explicit constructor here and not `@RequiredArgsConstructor` like below at `ContactSummary`?

File: backend/src/main/java/quarano/department/web/TrackedCaseController.java [hunk 3]
```diff
@@ -327,14 +342,22 @@
 				.build();
 	}
 
-	private Stream<TrackedCaseContactSummary> createSummaries(TrackedCase trackedCase) {
```
Comment (odrotbohm) [inline backend/src/main/java/quarano/department/web/TrackedCaseController.java:330]: Let's avoid unrelated renames.

Comment (jekutzsche) [reply backend/src/main/java/quarano/department/web/TrackedCaseController.java:330]: What do you mean? 
I think the renaming is good to make the sense of the method more concrete compared to the new method createDiaryEntrySummaries.

**Comment by marcoebbinghaus** [comment]
I think you are right and there is no reason for adding the constructor manually here. So I removed the constructor and added the Lombok annotation for that. Also I removed the unused messages field in TrackedCaseDiaryEntrySummary.

File: backend/src/main/java/quarano/department/web/TrackedCaseController.java (hunk 3 above)
Comment (marcoebbinghaus) [reply backend/src/main/java/quarano/department/web/TrackedCaseController.java:330]: Yes, could you please clarify? I don't get what you mean with unrelated names.

Comment (odrotbohm) [reply backend/src/main/java/quarano/department/web/TrackedCaseController.java:330]: Ahh, that makes a lot of sense then. The context wasn't quite obvious to me. Thanks for elaborating.

**Comment by odrotbohm** [comment]
Ready to take it from here. Would you mind squashing the commits into one and brush up the commit message elaborating a bit on the details of what's introduced here? I don't care about the individual commit steps but would rather be able to read a summary of the exposed API, important aspects of the implementation etc.

**Comment by odrotbohm** [comment]
That's polished and merged. Thanks!
